python/src/nanoarrow/_lib.pyx

```diff
@@ -84,3 +85,329 @@ def as_numpy_array(arr):
+            return self._ptr.format.decode("UTF-8")
+
+    @property
+    def name(self):
+        self._assert_valid()
+        if self._ptr.name != NULL:
+            return self._ptr.name.decode("UTF-8")
+        else:
+            return None
+
+    @property
+    def flags(self):
+        return self._ptr.flags
+
+    @property
+    def children(self):
+        self._assert_valid()
+        return CSchemaChildren(self)
+
+    def parse(self):
+        self._assert_valid()
+
+        cdef ArrowError error
+        cdef ArrowSchemaView schema_view
+
+        cdef int result = ArrowSchemaViewInit(&schema_view, self._ptr, &error)
+        if result != NANOARROW_OK:
+            raise ValueError(ArrowErrorMessage(&error))
+
+        out = {
+            'name': self._ptr.name.decode('UTF-8') if self._ptr.name else None,
+            'type': ArrowTypeString(schema_view.type).decode('UTF-8'),
+            'storage_type': ArrowTypeString(schema_view.storage_type).decode('UTF-8')
+        }
+
+        if schema_view.storage_type in (NANOARROW_TYPE_FIXED_SIZE_LIST,
+                                        NANOARROW_TYPE_FIXED_SIZE_BINARY):
+            out['fixed_size'] = schema_view.fixed_size
+
+        if schema_view.storage_type in (NANOARROW_TYPE_DECIMAL128,
+                                        NANOARROW_TYPE_DECIMAL256):
+            out['decimal_bitwidth'] = schema_view.decimal_bitwidth
+            out['decimal_precision'] = schema_view.decimal_precision
+            out['decimal_scale'] = schema_view.decimal_scale
+
+        return out
+
+cdef class CArray:
+    cdef object _base
+    cdef ArrowArray* _ptr
+    cdef CSchema _schema
+
+    @staticmethod
+    def Empty(CSchema schema):
+        base = CArrayHolder()
+        return CArray(base, base._addr(), schema)
+
+    def __init__(self, object base, uintptr_t addr, CSchema schema):
+        self._base = base,
+        self._ptr = <ArrowArray*>addr
+        self._schema = schema
+
+    def _addr(self):
+        return <uintptr_t>self._ptr
+
+    def is_valid(self):
+        return self._ptr.release != NULL
+
+    def _assert_valid(self):
+        if self._ptr.release == NULL:
+            raise RuntimeError("Array is released")
+
+    @property
+    def schema(self):
+        return self._schema
+
+    @property
+    def children(self):
+        return CArrayChildren(self)
+
+    def validate(self):
+        cdef CArrayViewHolder holder = CArrayViewHolder()
+
+        cdef ArrowError error
+        cdef int result = ArrowArrayViewInitFromSchema(&holder.c_array_view,
+                                                       self._schema._ptr, &error)
+        if result != NANOARROW_OK:
+            raise ValueError(ArrowErrorMessage(&error))
+
+        result = ArrowArrayViewSetArray(&holder.c_array_view, self._ptr, &error)
+        if result != NANOARROW_OK:
+            raise ValueError(ArrowErrorMessage(&error))
+
+        return CArrayView(holder, holder._addr(), self)
```

Comment:
It sounds a bit strange that "validate" returns a array view object.

(I would personally also add all properties of CArrayView like length, buffers, etc to the CArray class itself)